### ISSUE TITLE
Add function to report number of "hits" per individual search term.

### ISSUE THREAD

**Issue by mcguinlu** [ropensci/medrxivr#13]
It is often useful to be able to see the number of "hits" (records returned) by each individual element of the search, so that when designing the search strategy you can interogate which elements are influencing the returned records the most. So for example, if the search is:

``` r
topic1 <- c("dementia", "Alzheimer's")      # Combined with Boolean "OR"
topic2 <- c("lipids", "cholesterol")             # Combined with Boolean "OR"

query <- list(topic1,topic2)                        # Combined with Boolean "AND"

results <- mx_search(mx_snapshot(), query)
```

Then passing `query` to the proposed `mx_reporter()` function would return something like the below:

``` r
# Total number of records found by your search: XX

# Total topic 1 records: XX
# - dementia: XX
# - Alzheimer's: XX

# Total Topic 2 records: XX
# - lipids: XX
# - cholesterol: XX

```

**Comment by Jamesohare1** [comment]
Hey @mcguinlu. I've had a look at this issue and have implemented a solution. For the example above mx_reporter would output  the following:

```
Using medRxiv snapshot - 2020-12-18 00:37
Found 9 record(s) matching your search.

Total topic 1 records: 241
dementia :  170
Alzheimer's :  111

Total topic 2 records: 182
lipids :  73
cholesterol :  128
```

Just need to tidy up  a couple of things, and then can put in a pull request?

**Comment by mcguinlu** [comment]
Hi @Jamesohare1, 

Thanks for getting involved in this project, and for your patience on this issue/PR - I took an extended Christmas break and am only getting back up to speed now. I'll take a look at the PR sometime this week and get back to you with my full comments.

**Comment by Jamesohare1** [comment]
Hi @mcguinlu ,

Finally got around to putting in an updated pull request for this.  Feel free to give me a shout if you've any comments or questions.

Jim

**Comment by mcguinlu** [comment]
Closed in #23
